src/metabase/public_settings.clj

```diff
@@ -638,3 +638,9 @@
   (deferred-tru "Prefix for upload table names")
   :visibility   :authenticated
   :type         :string)
+
+(defsetting user-visibility
```

Comment:
I think this setting should maybe just go in `metabase.api.user` or a related namespace. This `public-settings` namespace is mostly for settings which are intended to have public visibility. It's gotten a bit polluted, and it's not a big deal, but might as well put this closer to the code that's using it